### PR TITLE
making inner classes public for mockito

### PR DESCRIPTION
**Pull request by joseblas**
for fixing pct issues.

### CODE REVIEW

**Comment by bitwiseman** [comment]
@joseblas Which issues are being fixed here?

**Comment by joseblas** [comment]
https://cloudbees.atlassian.net/browse/NGPIPELINE-740?atlOrigin=eyJwIjoiaiIsImkiOiJBODNFRkNENzA1RDE0MjE3QjA4Q0I4NDUyOUIzRTdCOCJ9
I think.

**Comment by car-roll** [comment]
@bitwiseman #192 bumped the parent pom and incorporated the changes @joseblas made in this PR

File: src/test/java/jenkins/branch/BranchCategoryFilterTest.java [hunk 1]
```diff
@@ -147,21 +147,21 @@ public void filter_specificCategory() throws Exception {
 
     }
 
-    private static abstract class TopLevelJob extends Job<TopLevelJob, TopLevelRun> implements TopLevelItem {
+    public static abstract class TopLevelJob extends Job<TopLevelJob, TopLevelRun> implements TopLevelItem {
```
Comment (bitwiseman) [inline src/test/java/jenkins/branch/BranchCategoryFilterTest.java:150]: If this is purely for mockito to not blow up on these, then internal should be okay. 

```suggestion
    static abstract class TopLevelJob extends Job<TopLevelJob, TopLevelRun> implements TopLevelItem {
```

File: src/test/java/jenkins/branch/BranchCategoryFilterTest.java [hunk 2]
```diff
@@ -147,21 +147,21 @@ public void filter_specificCategory() throws Exception {
...
 
         protected TopLevelJob(ItemGroup parent, String name) {
             super(parent, name);
         }
     }
 
-    private static abstract class TopLevelRun extends Run<TopLevelJob, TopLevelRun> {
+    public static abstract class TopLevelRun extends Run<TopLevelJob, TopLevelRun> {
```
Comment (bitwiseman) [inline src/test/java/jenkins/branch/BranchCategoryFilterTest.java:157]:
```suggestion
    static abstract class TopLevelRun extends Run<TopLevelJob, TopLevelRun> {
```

File: src/test/java/jenkins/branch/BranchCategoryFilterTest.java [hunk 3]
```diff
@@ -147,21 +147,21 @@ public void filter_specificCategory() throws Exception {
...
 
         protected TopLevelRun(@Nonnull TopLevelJob job) throws IOException {
             super(job);
         }
     }
 
-    private static abstract class MockSCMSource extends SCMSource {
+    public static abstract class MockSCMSource extends SCMSource {
```
Comment (bitwiseman) [inline src/test/java/jenkins/branch/BranchCategoryFilterTest.java:164]:
```suggestion
    static abstract class MockSCMSource extends SCMSource {
```

**Pull request closed without merging**